src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -25,6 +25,648 @@
+        """Get the timestamps for each frame.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of timestamps in seconds for each frame.
+
+        Notes
+        -----
+        This method extracts timestamps from the ScanImage TIFF file(s) for the selected channel.
+        It uses the mapping created during initialization to efficiently locate and extract
+        timestamps for each sample.
+        """
+        if self._times is not None:
+            return self._times
+
+        # Initialize array to store timestamps
+        timestamps = np.zeros(self._num_samples, dtype=np.float64)
+
+        # For each sample, extract its timestamp from the corresponding file and IFD
+        for sample_index in range(self._num_samples):
+
+            # Get the last frame in this sample to get the timestamps
+            frame_index = sample_index * self._num_planes + (self._num_planes - 1)
+            table_row = self._frames_to_ifd_table[frame_index]
+            file_index = table_row["file_index"]
+            ifd_index = table_row["IFD_index"]
+
+            tiff_reader = self._tiff_readers[file_index]
+            image_file_directory = tiff_reader.pages[ifd_index]
+
+            # Extract timestamp from the IFD description
+            description = image_file_directory.description
+            description_lines = description.split("\n")
+
+            # Use iterator pattern to find frameTimestamps_sec
+            timestamp_line = next((line for line in description_lines if "frameTimestamps_sec" in line), None)
+
+            if timestamp_line is not None:
+                # Extract the value part after " = "
+                _, value_str = timestamp_line.split(" = ", 1)
+                try:
+                    timestamps[sample_index] = float(value_str.strip())
+                except ValueError:
+                    # If parsing fails, use sample index / sampling frequency as fallback
+                    timestamps[sample_index] = sample_index / self._sampling_frequency
+            else:
+                # If no timestamp found, throw a warning and use sample index / sampling frequency as fallback
+                warnings.warn(
+                    f"No frameTimestamps_sec found for sample {sample_index}. Using calculated timestamp instead.",
+                    UserWarning,
+                )
+                timestamps[sample_index] = sample_index / self._sampling_frequency
+
+        # Cache the timestamps
+        self._times = timestamps
+        return timestamps
+
+    def get_num_planes(self) -> int:
+        """Get the number of depth planes.
+
+        For volumetric data, this returns the number of Z-planes in each volume.
+        For planar data, this returns 1.
+
+        Returns
+        -------
+        int
+            Number of depth planes.
+        """
+        return self._num_planes
+
+    def get_plane_extractor(self, plane_index: int) -> "ImagingExtractor":
```

Comment:
```suggestion
    def get_plane_extractor(self, plane_index: int) -> ImagingExtractor:
```

I don't think you need the quotes here

Reply:
Thanks.